### ISSUE TITLE
vim 9's popup features not working

### ISSUE THREAD

**Issue by xave** [kovidgoyal/kitty#7464]
**Describe the bug**
https://vim-jp.org/vimdoc-en/popup.html

`vim` has the following pop up features. They work properly in `iterm2` but do not work in `kitty`.

**To Reproduce**
Stock kitty config. Changed a color theme with `kitten themes` command. Otherwise, stock. `vim` and `tmux` configs picked up automatically as expected.

**Screenshots**
`vim` pop up menu (PMenu) as shown in `iterm2`

<img width="378" alt="Screenshot 2024-05-21 at 6 32 14 PM" src="https://github.com/kovidgoyal/kitty/assets/8461063/2b7ded00-f19b-4d1e-a7ff-1ee51890e4cf">

`vim` not showing pop up menu or side column in `kitty`

<img width="378" alt="Screenshot 2024-05-21 at 6 33 06 PM" src="https://github.com/kovidgoyal/kitty/assets/8461063/3501ccf6-d644-464a-bc6c-14fc9b571a70">

**Environment details**

```
kitty 0.34.1 created by Kovid Goyal
Darwin Kernel Version 23.5.0: Wed May  1 20:17:33 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6031 arm64
ProductName:            macOS ProductVersion:           14.5 BuildVersion:              23F79
Frozen: False
```

**Comment by kovidgoyal** [comment]
I strongly recommend switching to nvim, but if you really want to
continue using vim, see
https://sw.kovidgoyal.net/kitty/faq/#using-a-color-theme-with-a-background-color-does-not-work-well-in-vim

**Comment by xave** [comment]
Not sure why this is closed. These seem to me to be completely different issues.

**Comment by kovidgoyal** [comment]
Because this is not a bug in kitty. It's a bug in vim, that you most
likely can workaround by following the instructions in that FAQ.

**Comment by xave** [comment]
Could you expand on what the `vim` bug is? I did read your entire thread with Bram. And I added all of the hardcoded settings you recommend. They did nothing for this particular issue. If you have further insights, it would be appreciated.

https://github.com/fatih/vim-go/issues/1963

This is the functionality I am trying to get, loosely. The pop up balloon should pop up.

**Comment by xave** [comment]
Perhaps there is a specific `$TERM` setting I should have. You are more familiar with the nuances of this and thus could provide better insights.

**Comment by kovidgoyal** [comment]
No clue, I stopped using vim a while back, given all its issues. General
advice is, make sure TERM is xterm-kitty, and you have applied the settings
in the FAQ. Beyond that you are better of asking in the vim issue
tracker, they will know best what preconditions are needed for whatever
is responsible for that feature. There is nothing the terminal has to do
here, it's all application side code, vim needs to decide to draw or not
draw those popup windows and then draw them or not draw them. The popup
windows are themselves not actually "windows" they are just a
differently colored region of the screen completely indistinguishable
from anything else as far as the terminal emulator is concerned.

**Comment by xave** [comment]
Thanks for your response. I have found `vim` to be stable and reliable companion over the years. The big change for me is switching from `iterm2`, which I have been using for some time. So some growing pains are to be expected. Though, most of my setup does just work out of the box with `kitty`, so I am willing to do a bit more to make the switch.

I did try the `TERM=xterm-kitty` to no effect with respect to this scenario. I will investigate further.

**Comment by xave** [comment]
I figured it out. My particular workflow uses the popup balloons with an LSP client. I am using `nix`. The LSP part does not work unless I start the `nix-shell`. I did that and immediately got the popup. 

This means that my config worked with your system out of the box, without doing that hardcoding of parameters. It also means that it worked out of the box even without me setting the terminal to `xterm-kitty`. Just FYI for you. 

Thanks for creating such a performant terminal. I might just make the switch!
